src/hotspot/os/posix/os_posix.cpp

```diff
@@ -247,6 +247,25 @@ int os::create_file_for_heap(const char* dir) {
   return fd;
 }
 
+// Is a (classpath) directory empty?
+bool os::dir_is_empty(const char* path) {
+  DIR *dir = NULL;
+  struct dirent *ptr;
+
+  dir = opendir(path);
+  if (dir == NULL) return true;
+
+  // Scan the directory
+  bool result = true;
+  while (result && (ptr = readdir(dir)) != NULL) {
```

Comment:
This will call `os::readdir` when we can just call `::readdir` directly.

Reply:
Hi David , so do you suggest to use ::readdir instead ? What about opendir and closedir, should we use as well ::opendir and ::closedir ?

Reply:
Yes, sorry, all of them. Thanks.

Reply:
> Yes, sorry, all of them. Thanks.

I adjusted the 3 calls.